numba/tests/test_fancy_indexing.py

```diff
@@ -255,6 +255,19 @@ def check(arr, ind):
 
         #exceptions leak refs
         self.disable_leak_check()
+    
+    def test_newaxis(self):
+        def np_new_axis(a, idx):
+            return a[idx, np.newaxis, :]
+        
+        arr = np.random.randint(0, 10, (4,5,6))
+        idx = np.array([1,2,1])
+        py_func = np_new_axis
+        cfunc = njit(py_func)
+
+        expected = py_func(arr, idx)
+        got = cfunc(arr, idx)
+        np.testing.assert_equal(expected, got)
```

Comment:
Think it'd be a good idea to include more tests:
* Multiple `np.newaxis` in a single indexing call.
* `np.newaxis` with other involved indexes.
* `np.newaxis` in a `__setitem__` context.
* Use of `None` in place of `np.newaxis`.

Reply:
Resolved in [be5d0cc](https://github.com/numba/numba/pull/8513/commits/be5d0ccbbff8b33615fc6e2a6eb88c104fc5dbf2)